parts/k8s/on-restart.ps1

```diff
@@ -0,0 +1,93 @@
+$global:LogPath = "c:\k\on-restart.log"
+$global:HNSModule = "c:\k\hns.psm1"
+
+filter Timestamp { "$(Get-Date -Format o): $_" }
+
+function Write-Log ($message) {
+    $message | Timestamp | Tee-Object -FilePath $global:LogPath -Append
+}
+
+Write-Log "Entering on-restart.ps1"
+
+#
+# Stop services
+#
+Write-Log "Stopping kubeproxy service"
+Stop-Service kubeproxy
+
+Write-Log "Stopping kubelet service"
+Stop-Service kubelet
+
+#
+# Perform cleanup
+#
+Import-Module $global:HNSModule
+
+Write-Log "Cleaning up persisted HNS policy lists"
+Get-HnsPolicyList | Remove-HnsPolicyList
+
+$hnsNetwork = Get-HnsNetwork | Where-Object Name -EQ azure
+if ($hnsNetwork){
+    Write-Log "Cleaning up HNS network 'azure'..."
+
+    Write-Log "Cleaning up containers"
+    docker ps -q | ForEach-Object {docker rm $_ -f}
+
+    Write-Log "Removing old HNS network"
+    Remove-HnsNetwork $hnsNetwork
+
+    taskkill /IM azure-vnet.exe /f
+    taskkill /IM azure-vnet-ipam.exe /f
+
+    $cnijson = [io.path]::Combine("c:\k", "azure-vnet-ipam.json")
```

Comment:
because these were bug workarounds, I think we need to be able to track when mitigations were applied. Can you move this into a loop structure so that we get logs if the files were found and when they're deleted?

Reply:
done